src/simple-pg/transaction.js

```diff
@@ -0,0 +1,111 @@
+import {
+  isPlainObject,
+  decodeArgs,
+  wrap,
+  unwrap,
+  finalize,
+  merge,
+  wrapObject,
+  slice,
+  encodeGraph,
+  isEmpty,
+  isRange,
+  decodeGraph,
+} from '@graffy/common';
+import { selectByArgs, selectByIds } from './sql/select';
+import { put, patch } from './sql/index.js';
+import { linkChange } from './link/index.js';
+import pg from './pool.js';
+import { format } from '@graffy/testing';
+import debug from 'debug';
+const log = debug('graffy:pg:dbRead');
+
+export const dbRead = async (rootQuery, opts, store) => {
+  const idQueries = {};
+  const promises = [];
+  const results = [];
+  const { id } = opts;
+  async function getByArgs(args, subQuery) {
+    let result = await pg.select(selectByArgs(args, opts));
+    result = result.map(([object]) => [{ ...object, keys: object[id] }]);
```

Comment:
I think you meant `$key` here, not `keys`. However this approach (setting `$key: object[id]`) is only correct for getById and not for getByArgs.

GetByArgs is complicated. It needs a $key and a $ref, and the $key is not so easy to calculate. I would recommend dropping getByArgs in the first version; if you still want to build it, read on...

Remember:
1. `$key` is just a fancy way of specifying object keys. Like `{ foo: ... }` is the same as `[{ $key: foo, ... }]`.
2. In Graffy read providers, the shape of the result returned must match the shape of the query received. This means the keys (and therefore `$key`s) in your response must match* those in the query you are responding to.

* match means equal, except in range queries ($first etc.) where it means "within the range".

Say your query is:
```js
// Get the name of user with email foo@x.com
user: [
  {
    $key: { email: 'foo@x.com' },
    name: true
  }
]
```

Your provider must return:
```js
user: [
  {
    $key: { email: 'foo@x.com' } // See, it matches
    $ref: ['user', 3432] // The "canonical" path of this object
    name: 'Alice' // The requested data
  }
]
```

For a range query (e.g. "get the first 10 users of tenant A, sorted by creation date") $key gets a bit more complicated still.

Reply:
I'm gonna ignore the `getByArgs` implementation for now. Let the function there cause I want to know all the store syntaxes that lead to the function got hit.

Reply:
Let's remove it from this PR for now. Please also remove other unused functions. We can always add them back later, as we have a reference in `pg`)